### ISSUE TITLE
Add ability to specify multiple media use term IDs

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#77]
Islandora now supports multiple media use terms. It should be possible to specify multiple values in the `media_use_tid` config option.

**Comment by mjordan** [comment]
May need to include the first term ID in the initial PUT request then add additional term IDs via subsequent PATCHs.

**Comment by mjordan** [comment]
Also, add a check for media use terms in the CSV that correspond to media created by Contexts. A general warning issued by `--check` could identify anything other than "Original File", since it's safe to assume that Contexts that create derivatives use "Original File" as the source media use. Possible wording for this warning could be "Islandora Workbench has detected Media Use terms in your CSV in addition to 'Original File' that might conflict with derivative media (e.g., 'Thumbnail', 'Service File')."

**Comment by dara2** [comment]
Hi @mjordan - I tested this but am not seeing what I was expecting (Media Use is still showing as just "Original File" instead of "Original File, Thumbnail Image"). 
My .yml file includes this line: `media_use: "Thumbnail Image | Original File"`
I run the import, and then I look at the Media tab for an image and I see this:
![Screen Shot 2021-03-18 at 9 56 33 AM](https://user-images.githubusercontent.com/72275887/111637983-4a094180-87d0-11eb-9acd-276bf0c70b03.png)
NOTE: We turned off the action to generate thumbnail derivatives. So no thumbnail is created at all.
We were expecting to see something like this:
<img width="1113" alt="screen_shot_2021-03-18_at_9 50 12_am" src="https://user-images.githubusercontent.com/72275887/111638333-9a809f00-87d0-11eb-8d34-ec97a6b925b5.png">

**Comment by mjordan** [comment]
You need to use term IDs in the `media_use_tid` setting, not term names, e.g. `media_use_tid: 17|18`.

We can make term names work, but I'd like to do that as a separate step.

**Comment by dara2** [comment]
Got it - thanks! I will try that.

**Comment by dara2** [comment]
Hi @mjordan - So, that works as expected. However...now we're running into another issue. Is there any way to apply these multiple term IDs ONLY to certain file types? We want PDFs to stay only as "Original File" (so that we can still generate those thumbnail derivatives using Drupal actions). Perhaps this is a new issue/request - if so, let me know and I can open one. :)

**Comment by dara2** [comment]
I guess the question is, could we add a column `field_media_use` to the CSV and put the Media Use term or terms in there on a per-object basis (so some objects have two, some just have one)? (Rather than declaring the Media Use terms in the .yml file.)

**Comment by mjordan** [comment]
Correct, the media use terms apply to all files. Providing a way to do it at the CSV record level is the way to go. 

One thing though - I don't want to mix using node fields and media fields in one CSV because it would complicate the way that Workbench gets field-level configuration from Drupal. For this situation, where we already have configuration option, I'd rather reuse that configuration name in the CSV as a 'reserved' field. In other words, the column name where you'd put the media use terms wouldn't be `field_media_use`, it would be `media_use_tid`. The value in the configuration option would still apply to all CSV records, unless there was a value in a record. Workbench already uses this pattern for a couple of other configuration settings.

This weekend is tied up with some other things so I probably won't get to this then but it's not that complicated.

**Comment by dara2** [comment]
That sounds perfect, thanks @mjordan !

**Comment by dara2** [comment]
One other note: I tried adding `media_use_tid: 5` to the .yml file (term ID for our Original File) and I got this error during the --check:
```
Traceback (most recent call last):
  File "./workbench", line 899, in <module>
    check_input(config, args)
  File "/Users/Dara 1/islandora_workbench/workbench_utils.py", line 829, in check_input
    validate_media_use_tid(config)
  File "/Users/Dara 1/islandora_workbench/workbench_utils.py", line 1262, in validate_media_use_tid
    media_use_terms = config['media_use_tid'].split(config['subdelimiter'])
AttributeError: 'int' object has no attribute 'split'
```
Previously (even though it wasn't really necessary) I was able to specify my Media Use term ID in the .yml file if I wanted to. Now it seems like it doesn't like it if I just have one term ID? It's fine when I leave out the `media_use_tid` altogether, and fine when I do `media_use_tid: 5|8`.
This is an instance of Drupal 8.9.14-dev.

**Comment by mjordan** [comment]
This ^ is simple bug I will fix, thanks for catching it.

**Comment by mjordan** [comment]
Closed with b37f3539a17559f9a68247114972b92543fbd376.
